src/Repository/PlaylistSlideRepository.php

```diff
@@ -114,6 +113,57 @@ public function updateRelations(Ulid $playlistUlid, ArrayCollection $collection)
         }
     }
 
+    public function updateSlidePlaylistRelations(Ulid $slideUlid, ArrayCollection $collection)
+    {
+        $slideRepos = $this->entityManager->getRepository(Slide::class);
+        $slide = $slideRepos->findOneBy(['id' => $slideUlid]);
+        if (is_null($slide)) {
+            throw new InvalidArgumentException('Slide not found');
+        }
+
+        $playlistRepos = $this->entityManager->getRepository(Playlist::class);
+
+        $this->entityManager->getConnection()->beginTransaction();
+        try {
+            // Remove all existing relations between slides and current playlist.
+            $entities = $this->findBy(['slide' => $slide]);
+            foreach ($entities as $entity) {
+                $this->entityManager->remove($entity);
+            }
+
+            if (0 == count($collection)) {
+                $playlistSlide = $this->findOneBy(['slide' => $slideUlid]);
+                if ($playlistSlide) {
+                    $this->entityManager->remove($playlistSlide);
+                    $this->entityManager->flush();
+                }
+            }
+
+            foreach ($collection as $entity) {
+                $playlist = $playlistRepos->findOneBy(['id' => $entity->playlist]);
+                if (is_null($playlist)) {
+                    throw new InvalidArgumentException('Playlist not found');
+                }
+
+                // Create new relation.
+                $ps = new PlaylistSlide();
+                $ps->setPlaylist($playlist)
+                    ->setSlide($slide)
+                    ->setWeight(0);
```

Comment:
Shouldn't the weight change according to where the slide should occur in the playlist? 
So you cannot just remove all the existing relations and just make them anew with weight 0. This will change playlist order.